### ISSUE TITLE
[management-api] Timing Issue with Rest-Api-Management

### ISSUE THREAD

**Issue by insertmike** [gravitee-io/issues#3671]
### Abstract

This issue is stating a timing issue with the gravitee-rest-api-management, occuring when a client executes multiple consecutive api calls in a short period of time.

### Context
I am currently implementing an internal SDK for the Gravitee API Management. To ensure the accuracy of my code I am writing unit test for every feature I implement such as: addApi, removeApi, listPlans and other. I have mentioned unexpected behaviour of the rest-api-management when the unit test framework is ran on it's own and I believe that this is because there is multiple api calls performed in a short period of time. 

### Current Behavior

I am experiencing different behaviour when I leave the unit test framework to run the tests on it's own and when I run them one by one. Let's take for example the following 3 consecutive operations:

1. Check if API with PATH exists
1.1 Delete it if exists
2. Create API with PATH
3. Delete API with PATH

All of this is working perfectly fine when running the tests manually and fails when the unit test framework runs on it's own. This all can be verified by following the Steps to Reproduce section.

### Steps to Reproduce
To demonstrate the strange behaviour of Gravitee I have created a simple python file, which has implemented 3 functionalities of the gravitee-rest-api-management:
- createApi() 
- getApiId()
- removeApi()

And 2 methods for demonstration of the behaviour:
- fastLoop() : Performing CreateApi, getApiID, removeID consecutive in a loop with 100 iterations. Usually fails on first to second iteration. 
<p align="center">
   <img src="https://user-images.githubusercontent.com/45242072/80594376-8a7e9600-8a2b-11ea-9eb3-522c7b3b9abb.png" width="600"  height="auto">
</p>

2. slowLoop(): Performing the exact same as fastLoop, identical code  but with 3 seconds sleep in between the calls. Does not fail.
<p align="center">
   <img src="https://user-images.githubusercontent.com/45242072/80594429-a6823780-8a2b-11ea-8d08-fabc7a9dc3b4.png" width="600"  height="auto">
</p>

To run the code below on your machine:
1. Ensure you have Python installed by running `python --version`
2. Ensure you have PIP installed by running `python -m pip --version`
3. Ensure you have python `requests` library installed by running `pip install requests`
4. Ensure you have Gravitee up and running ( Elastic Search, Gravitee-Api-Gateway, Gravitee-Api-Rest-Api-Management )
5. Configure the CONFIG variables in the python code below as are on your system. The variables are assigned to the default ones when gravitee is installed.
6. Run the code with `run_fast_loop` assigned to True and to False to experience the Fast Loop Behaviour and the Slow Loop Behaviour

```python
import requests
from requests.auth import HTTPBasicAuth 
from time import sleep
# --------- CONFIGURATION ---------
# 1. Ensure Python and Pip is installed by running python --version and pip --version
# 2. Ensure requests library is installed by running 'pip install requests' 
# 3. Ensure Gravitee Gateway and Management-API are up and runnning 
# 4. Configure the following variables which are dependent on your gravitee installation:
api_management_url = 'http://localhost:8083/management'
management_username = 'admin'
management_password = 'admin'
# *** When run_fast_loop is False, slow loop runs, and the opposite *** 
# FAST LOOP -> No sleep between the Api calls
# SLOW LOOP -> 3 Seconds sleep in between the Api calls 
# Loops code is identical apart from the sleep 
run_fast_loop  = True
run_slow_sloop = not run_fast_loop
# ---------------------------------
​
# This is the path of the API which will be deleted and inserted in the code below
context_path = '/testPath'
​
​
def createApi(path):				
		res = requests.post(api_management_url + '/apis', auth=HTTPBasicAuth(management_username, management_password),
		json={'name': 'Test', 'version' : '1.0', 'description': 'Test Description', 'contextPath': path, 'endpoint':'localhost:81'})
		if res.status_code != 201:			
			print('Failed to create api with contextpath: ', path)
			quit()
		api_id = res.json()['id']		
		return api_id
		
		
def removeApi(api_id):
	res = requests.delete(api_management_url + "/apis/" + api_id, auth=HTTPBasicAuth(management_username, management_password))
	if res.status_code == 204:				
		return True
	else:
		print('Api failed to remove with status code of: ', res.status_code)
		quit()		
​
def getApiId(context_path):
	res = requests.post(api_management_url + "/apis/_search?q=" + context_path, auth=HTTPBasicAuth(management_username, management_password))
	
	if len(res.json()) < 1:		
		return None
	return res.json()[0]['id']
​
# ----- REMOVE API CONTEXT PATH IF ALREADY EXIST -----
print('Initial check if context path: ', context_path, ' already exist')
​
api_id = getApiId(context_path)
​
if api_id:
	print('Initial check finished. Api exists. Api ID: ',api_id)
	if not removeApi(getApiId(context_path)):		
		quit()
	print('Deleted API ID: ', api_id)
else:
	print('Initial check finished. Api does not exist, proceeding to for loop.')
# ----- REMOVE API CONTEXT PATH IF ALREADY EXIST -----
​
print('------  ENTERING FOR LOOP ------')
def fastLoop():
	print('Entering FAST loop. That means that we have no sleeping in between the api calls')
	# Everything in this for loop should run smoothly
	for _ in range(1,100):
		api_id_from_creation = createApi(context_path)	
		if None == api_id_from_creation:		
			print('************')
			print('Failed to create api with context_path: ', context_path)
			print('************')
			quit()		
		print('Created api with context path: ', context_path, 'API ID: ', api_id_from_creation)		
		api_id_from_get = getApiId(context_path)		
		if None == api_id_from_get:
			print('************')
			print('Failed to get api context path:', context_path)
			print('************')
			quit()
		print('Got api by its context path: ', context_path, 'API ID: ', api_id_from_get)		
		if not removeApi(api_id_from_get):
			print('************')
			print('Failed to remove api id: ', api_id_from_get)
			print('************')
			quit()
		print('Deleted API ID: ', api_id_from_get)				
		print('-------------------- ITERATION:', _, 'finished')
		
def slowLoop():
	print('Entering SLOW loop. That means that we have 3 seconds sleep in between the api calls')
	# Everything in this for loop should run smoothly
	for _ in range(1,100):
		sleep(3)		
		api_id_from_creation = createApi(context_path)					
		if None == api_id_from_creation:		
			print('************')
			print('Failed to create api with context_path: ', context_path)
			print('************')
			quit()		
		print('Created api with context path: ', context_path, 'API ID: ', api_id_from_creation)		
		sleep(3)		
		api_id_from_get = getApiId(context_path)				
		if None == api_id_from_get:
			print('************')
			print('Failed to get api context path:', context_path)
			print('************')
			quit()
		print('Got api by its context path: ', context_path, 'API ID: ', api_id_from_get)		
		sleep(3)		
		if not removeApi(api_id_from_get):
			print('************')
			print('Failed to remove api id: ', api_id_from_get)
			print('************')
			quit()
		print('Deleted API ID: ', api_id_from_get)				
		print('-------------------- ITERATION:', _, 'finished')	
if run_fast_loop:
	fastLoop()
else:
	slowLoop()
```

### Your Environment
Not environment dependent.

**Comment by brasseld** [comment]
duplicate of https://github.com/gravitee-io/issues/issues/3671 ?

**Comment by jhaeyaert** [comment]
Seems not a duplicate of #4130 which is related to a gateway (re)deployment issue. This one seems only be related to the management api.
Unfortunately, I executed the given script on the v3.1.x (by the way, thx for this script 👍 ) and couldn't make it fail in fast mode.

I suspect this script failed on v1.x isn't it ? (I had to update the management api url to add default org & env which were inexistant in v1.x)
